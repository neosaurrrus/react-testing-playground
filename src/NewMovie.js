import React, { Component } from 'react'
import MovieForm from './MovieForm'

export default class NewMovie extends Component {
    render() {
        return (
            <div>
                 <h1 data-testid='NewMovie-heading'>Enter a New Movie</h1>
                 <MovieForm data-testid="movie-form"/>
            </div>
        )
    }
}
