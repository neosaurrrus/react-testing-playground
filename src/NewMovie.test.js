import React from 'react'
import { render, cleanup, fireEvent} from 'react-testing-library'; //Added FireEvent from React Testing Library
import NewMovie from './NewMovie';

afterEach(cleanup)

test('<NewMovie>', () => {
  const {debug, getByTestId, queryByTestId, container} = render(<NewMovie/>)
  debug()
  const NewMovieHeading = getByTestId('NewMovie-heading')
  expect(NewMovieHeading.textContent).toBe("Enter a New Movie")
  expect(NewMovieHeading.tagName).toBe("H1")
  //Check for Movie form
  
  expect(container.firstChild).toMatchSnapshot();
});