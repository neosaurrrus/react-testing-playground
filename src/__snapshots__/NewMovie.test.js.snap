// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewMovie> 1`] = `
<div>
  <h1
    data-testid="NewMovie-heading"
  >
    Enter a New Movie
  </h1>
  <div>
    <form
      data-testid="movie-form"
    >
      <label
        for="title"
      >
        Title
      </label>
      <input
        id="title"
        type="text"
      />
      <button
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
</div>
`;
